### PR TITLE
Lts upgrade

### PR DESCRIPTION
**Pull request by dfithian**
Uses stack + hpack

I presume one very controversial thing I did was to move `oauth_token` into `workflowParams`. I came to this conclusion by skimming through the [RFC](https://tools.ietf.org/html/rfc5849) and deciding that since I only ever saw `oauth_token` in conjunction with either `oauth_callback` or `oauth_verifier` that they should go together. I'd be perfectly happy to instead pass in some function of the settings telling it whether or not to include `oauth_token` for a given request. Whatever the conclusion, the service I'm integrating to specifically does **NOT** want the `oauth_token` so that was the motivation.

Another thing that travis did was call `cabal test` but since there are no tests I made this just `stack build`.

@mossprescott @ystael

### CODE REVIEW

**Comment by dfithian** [comment]
@tel Doesn't look like travis is set up to me, or I don't have permissions. When you get a chance can you point me to it or create the build and give me permissions to see it? I can also add a badge to the readme.

**Comment by dfithian** [comment]
Other thing is that crypto-random is deprecated, but I think it still works okay for now.

https://github.com/vincenthz/hs-crypto-random

File: oauthenticated.cabal [hunk 1]
```diff
@@ -1,87 +1,118 @@
-name:                oauthenticated
-version:             0.1.3.4
-synopsis:            Simple OAuth for http-client
+-- This file has been generated from package.yaml by hpack version 0.20.0.
```
Comment (mossprescott) [inline oauthenticated.cabal]: Probably just delete this file from the repo since it's now generated at build time?

Comment (dfithian) [reply oauthenticated.cabal]: I think some people building with cabal will still need it?

File: examples/oauth-authenticate.hs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+data App = App
+  { appRng     :: MVar Crypto.SystemRNG
+  , appManager :: Client.Manager
+  , appServer  :: OAuth.Server
+  , appCred    :: OAuth.Cred OAuth.Client
+  }
+
+signRequest :: App -> Client.Request -> IO Client.Request
```
Comment (mossprescott) [inline examples/oauth-authenticate.hs]: Can this be something like `StateT SystemRNG IO Client.Request` and take the rng out of App? Not a big deal, but bringing MVar into it seems unnecessary for an example.

Comment (dfithian) [reply examples/oauth-authenticate.hs]: 👍

**Pull request merged**